types/index.d.ts

```diff
@@ -10,16 +10,18 @@
  * Adding key augmention is tracked here: https://github.com/Microsoft/TypeScript/issues/12754
  */
 
-import {StyleSheet} from 'react-native';
+import {ImageStyle, StyleSheet, TextStyle, ViewStyle} from 'react-native';
 
 export = EStyleSheet;
 
 declare namespace EStyleSheet {
     type AnyObject<T = {}> = T & {[key: string]: any};
     type Event = 'build';
+    type ExtendedStyle<T> = { 
+        [K in keyof T]: ViewStyle | TextStyle | ImageStyle | string | number | AnyObject
+    }
 
-
-    export function create<T>(styles: AnyObject<T>): AnyObject<T>;
+    export function create<T>(styles: ExtendedStyle<T>): AnyObject<T>;
```

Comment:
Here I was getting an error " AnyObject is not a generic. I am using typescript. By changing the this line to the one below, it all worked. Thank you for helping with this!
```suggestion
    export function create<T>(styles: ExtendedStyle<T>): AnyObject;
```

Reply:
> Here I was getting an error " AnyObject is not a generic. I am using typescript. By changing the this line to the one below, it all worked. Thank you for helping with this!

I think its because you get type AnyObject from another module. I have `type AnyObject<T = {}> = T & {[key: string]: any};` above